### DISCUSSION TITLE
v1.21.14

### DISCUSSION THREAD

**Discussion by skovatch** [Ascoware/get-iplayer-automator#380] [Announcements]
- itv: Fix parsing of itv.com/hub/shows, which broke with an ITV Hub revamp

<hr /><em>This discussion was created from the release <a href='https://github.com/Ascoware/get-iplayer-automator/releases/tag/v1.21.14'>v1.21.14</a>.</em>

**Comment by SOLOSAIR** [comment]
Many thanks for update.  Coronation Street ITV success. However, Bradley & Barney Walsh: Breaking Dad (also already in PVR list) of ITV was not recognised and was not added by series link. Went to ITV website and tried 'use current webpage' and 'process ID' buttons but after pressing 'start' status is stuck on 'Processing...' and does not progress.

**Reply by skovatch** [reply]
Ugh... ITV changes triggered another bug. When parsing a show's page for episode URLs I assumed that the series' URL and episode URLs had the same base path. This lets the listings scanner skip over promotional entries like "If you like this, why not try..." shows. That assumption is no longer valid, so now I have to see if I already cached a program with the episode URL. NOT ideal, but it ensures I don't miss any programs.

**Reply by skovatch** [reply]
There's a separate issue in that if an ITV show isn't found in the cache, get_iplayer's extended search won't work because it only supports BBC. In that case the status string needs to be updated with "Not in cache".
